vim-patch:8.0.1786,8.1.{851,1308,1309,1579}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.0851: feedkeys() with "L" does not work properly

Problem:    feedkeys() with "L" does not work properly.
Solution:   Do not set typebuf_was_filled when using "L". (Ozaki Kiichi,
            closes vim/vim#3885)
https://github.com/vim/vim/commit/8d4ce56a19ed14d13332f94ad592fff2d9a715d5

janlazo commented on src/nvim/testdir/test_timers.vim:
```diff
@@ -314,4 +313,30 @@ func Test_restore_count()
+  set ut=200
+  call test_garbagecollect_soon()
+  " skipped: Nvim does not support test_override()
+  return
```
`throw` won't work here because of `_nocatch_` in function name.

justinmk replied:
we might as well skip `test_garbagecollect_soon` then.

janlazo commented on src/nvim/api/vim.c:
```diff
@@ -241,15 +241,15 @@ void nvim_feedkeys(String keys, String mode, Boolean escape_csi)
   ins_typebuf((char_u *)keys_esc, (remap ? REMAP_YES : REMAP_NONE),
-      insert ? 0 : typebuf.tb_len, !typed, false);
+              insert ? 0 : typebuf.tb_len, !typed, false);
+  if (vgetc_busy) {
```
8.1.0815 should comes first but it's a pain to port it now.

janlazo committed:
vim-patch:8.0.1786: no test for 'termwinkey'

Problem:    No test for 'termwinkey'.
Solution:   Add a test.  Make feedkeys() handle terminal_loop() returning
            before characters are consumed.
https://github.com/vim/vim/commit/b2ac14c0b5e23f8ab97c5c784bcd83e13ba8ded3

janlazo committed:
vim-patch:8.1.1579: dict and list could be GC'ed while displaying error

Problem:    Dict and list could be GC'ed while displaying error in a timer.
            (Yasuhiro Matsumoto)
Solution:   Block garbage collection when executing a timer.  Add
            test_garbagecollect_soon(). Add "no_wait_return" to
            test_override(). (closes vim/vim#4571)
https://github.com/vim/vim/commit/adc6714aac20f5462a0ecec50ab4806b2f3ab0db

janlazo committed:
vim-patch:8.1.1308: the Normal highlight is not defined when compiled with GUI

Problem:    The Normal highlight is not defined when compiled with GUI.
Solution:   Always define Normal. (Christian Brabandt, closes vim/vim#4072)
https://github.com/vim/vim/commit/f90b6e03a983b62b66564fc449e32724d6456769

janlazo committed:
vim-patch:8.1.1309: test for Normal highlight fails on MS-Windows GUI

Problem:    Test for Normal highlight fails on MS-Windows GUI.
Solution:   Skip the test for MS-Windows GUI.
https://github.com/vim/vim/commit/6b528fa062a5ac6bb5d8bd3abc26f32c65691d00
